### ISSUE TITLE
panasonic AW-RP150 ptz controller

### ISSUE THREAD

**Issue by fdebeer** [bitfocus/companion-module-requests#355]
support for the Panasonic AW-RP150 and the AW-RP130 PTZ controller.

New features with these controllers, like timed zoon and pan. (change preset in a given number of seconds.)
would be nice to have this under a button

**Comment by McHauge** [comment]
do you have any documentation on the API? you can currently control most ptz's directly so what specifically are you trying to access with the controller? to be camera independent?

**Comment by fdebeer** [comment]
i control the camera's with companion, but when i select a preset, i want also select the camera on the controller, so it's easy to adjust the camerapreset without selecting the camera first. so selecting a preset on cam1, also switch the controller to cam 1.
this is possible for the RP120 ans RP 50, but not for the new controllers. 
also it looks possible to connect the pc via IP to the controller to control the camera's connected via RS422.
found some documentation for this on https://eww.pass.panasonic.co.jp/pro-av/support/content/guide/DEF/RP50_120/RemoteControllerInterfaceSpecifications-E.pdf

hope you can do somethiong with this. for me it's abracadabra...

**Comment by McHauge** [comment]
Looks very doable, and very close to the PTZ's same structure, and even commands are the same for a lot of it, I'll most likely give it a look when I'm done with the massive update to the PTZ module. but I might need remote access to a controller, and I don't think we have one on loan atm.

**Comment by McHauge** [comment]
another note have you actually tried using the current module and seen if som stuff works or not, it's on the same protocol, so it should at least do something

**Comment by fdebeer** [comment]
connecting companion to the RP 60 or 150 is not working. no responce. 
Dit not try the RP50 or 120 yet, but there are modules for these in companion, so i think they wil work to select the camera

**Comment by McHauge** [comment]
Yes there are a module for the RP50 and 120, but did you try to use that module, as an example add the module select the model RP50 and give it the ip from you RP60 and see what happenes

**Comment by fdebeer** [comment]
i'm freelance. dont have the equipment here. wil have to wait for the next job, i'm afraid
tried to connect a RP150 with the RP120 module, but got no responce from the RP150. status 'null'

**Comment by McHauge** [comment]
even though the status is "null" that does not necessarily determine if action works or not, that's what you should test by just adding one of the buttons and see if it reacts to the commands. 

I just looked at the code, there is nowhere, where it determines what status to show, the only status change in the entire code is to show "null".... also I don't really like the way you chose what model you have and the fact that you can't change it after adding the module, it does not really match the way other modules operate.

in short, the "status" field in this module means absolutely nothing...

**Comment by fdebeer** [comment]
> in short, the "status" field in this module means absolutely nothing...

didn't know that. i learn every day... :)
Hope to sit on a job soon again. it's a bit slow now with the lockdown and so here...will try next time

**Comment by K9-Tim** [comment]
Hey there I'm interested in that topic too. I got a RP150 here, so I'll test it and send you my results. ;)

**Comment by McHauge** [comment]
@k9-tim lovely, would you be able to test the current module, just to see if something happens after all, even while selecting the "wrong" model

**Comment by fdebeer** [comment]
tested it yesterday with the RP 60. the module for rp 120 / 50 works with this controllers for selecting kamera or groups. probably  it also works for the RP150, as the protocol for selecting kamera's is the same.

but stil, the RP150 has more settings and options than the other controllers, so maybe a seperate module for the RP150?

**Comment by McHauge** [comment]
I don't think a new module, I feel like adding the support for different models, is the way to go, but currently, I'm working on the PTZ module, I plan on (if I can), reuse as much as possible for this after. since they are very close in the protocol, some of the commands are even identical.

**Comment by fdebeer** [comment]
> I don't think a new module, I feel like adding the support for different models, is the way to go, but currently, I'm working on the PTZ module, I plan on (if I can), reuse as much as possible for this after. since they are very close in the protocol, some of the commands are even identical.

It's the RP150 with some new options. like setting a time for a move between presets. other controllers don't have this.

**Comment by McHauge** [comment]
yes, but this (if possible) would also add feedbacks and variables to the older controllers

**Comment by fdebeer** [comment]
> yes, but this (if possible) would also add feedbacks and variables to the older controllers

read your facebookpost. going to test this on monday. Thanks! great work

**Comment by reykroona** [comment]
> Yes there are a module for the RP50 and 120, but did you try to use that module, as an example add the module select the model RP50 and give it the ip from you RP60 and see what happenes.
> 
> If you can a screenshot would be nice of the instance config page.

Hi, I stumbled upon this today testing out this module with an AW-RP60 and the new scheduling/trigger feature. Ultimately I'd like to have our ATEM change AUXs to match the camera selected on the RP60.

I found I can create a button in Companion to change the camera selection on the RP60 which works well, however the Camera Number variable doesn't get updated when it's changed on Companion or on the RP60, I believe it's because only the RP50 supports the response from the XCN:01:[Data] command according to the newer documentation.

It looks like they did add a new confirmation command XQC:01, I put this URL in my browser with the RP60 IP address and it returned the selected camera. I guess maybe it needs to setup a listener or a timer or something to send the request? Not exactly sure, I'm only an amateur programmer.

I was thinking about seeing if I could get something working, but sounds like you are much more able than me to figure this out. However, I'm happy to help test where I can. I have access to an AW-RP60, an AW-HE130 and AW-UE150.

Thanks! Look forward to helping out where I can.

Rey

**Comment by McHauge** [comment]
@reykroona about the variables in the current version of the module. They are not controlled by feedback from the controller, but instead when you press an action the update to match that button. I might call it worse than no feedback since might misinterpret it as real feedback around what the controller is doing, and giving you a fake sense of confidence.

Also if one of you have the option to set up a controller so that I can reach it, over the internet or via a TeamSpeak/remote desktop. that would make dev work way more simple :) Currently, I don't have a controller to test on so I'm a bit limited in how much freedom there is while working on it, as with the PTZ I can test most of it live while making changes.

**Comment by reykroona** [comment]
Ah I see. Interesting. 

I think I could probably bring the controller home and get it on the internet on Sunday and leave it on for the week. Sunday’s we need to use it so I’d have to take it back for the day. But otherwise it’s not doing much. The cameras are mounted so I probably won’t be able to take one of them home. 

I think it is just accessed through port 80, that sound right?

**Comment by McHauge** [comment]
@josephdadams would it be posible for you to transfer this to the Panasonic-camera-controller module

**Comment by McHauge** [comment]
@reykroona That would be awesome and would help out a lot if possible. But currently, I'm a bit full, and I might not look too much add it for at least 1 or 2 weeks
